### DISCUSSION TITLE
Support for opening files with line offset

### DISCUSSION THREAD

**Discussion by jellelicht** [magit/with-editor#120] [Ideas]
I’m not certain what the feature is called, but at least vim and nano support the following notation:
`<editor> +17 <filename>`

Would it be possible for with-editor to also support this mechanism?

**Comment by tarsius** [comment]
Emacs supports that too, see `emacs --help`. Are you sure you need `with-editor` to support it as well? What for?

**Reply by jellelicht** [reply]
I like eshell, so I run the `(add-hook ‘eshell-mode-hook ‘with-editor-export-editor)` snippet. 
I have a command (`guix edit some-package-name`) that uses the + notation with the current value of `$EDITOR` to open a file; right now, running that command in my eshell leads to an empty buffer being opened with the name `+87` (or any other line number) instead of the right file on the right line, which is not what I would be interested in.

**Reply by tarsius** [reply]
A useful command indeed.

I looked at posix in hope for some guidance but couldn't find anything. E.g., if a file `file+10` exists, should that be opened instead of `file` and then jumping to line 10?

I'll have to implement that for the "sleeping editor", which is used when `emacsclient` cannot be used, but when `emacsclient` is used, I was hoping this would just work. But it turns out that (even with `with-editor` is not involved at all) `emacsclient file+10` does not work. One has to use `emacsclient +10 file` instead. It might not be a good idea for `with-editor` to change how `emacsclient` treats the former. I didn't look at the `emacs --help` output closely enough before; that doesn't support `file+10` either! You might want to ask on emacs-devel why `emacs`/`emacsclient` don't support that; there clearly are tools that expect that to work.

**Reply by jellelicht** [reply]
Just to be clear, the thing I’m referring to is only the `$EDITOR +73 filename` notation. AFAICS `$EDITOR filename+73` is something else entirely, and not something I’m greatly bothered by, at least currently.

More concretely, I think my particular feature request would be fulfilled by adding support for the `+linum filename` notation to the “sleeping editor”.

**Reply by tarsius** [reply]
Ah, sorry, I didn't reread you initial message when replying to the second, and somehow misremembered what you asked for. Yes, I think what you actually asked for makes sense and should be doable.

**Reply by tarsius** [reply]
I've implemented this now.

**Reply by jellelicht** [reply]
Thank you @tarsius , much appreciated. Do you have any estimation on when you expect to make a tagged release again?

**Reply by tarsius** [reply]
Um, January was supposed to be Release Month. Looks more like* February now.

(*) *All* my packages need a new release. :|
